src/main/java/org/opensearch/flowframework/transport/DeprovisionWorkflowTransportAction.java

```diff
@@ -103,46 +100,25 @@ public DeprovisionWorkflowTransportAction(
 
     @Override
     protected void doExecute(Task task, WorkflowRequest request, ActionListener<WorkflowResponse> listener) {
-        // Retrieve use case template from global context
         String workflowId = request.getWorkflowId();
-        GetRequest getRequest = new GetRequest(GLOBAL_CONTEXT_INDEX, workflowId);
+        GetWorkflowStateRequest getStateRequest = new GetWorkflowStateRequest(workflowId, true);
 
         // Stash thread context to interact with system index
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-            client.get(getRequest, ActionListener.wrap(response -> {
+            client.execute(GetWorkflowStateAction.INSTANCE, getStateRequest, ActionListener.wrap(response -> {
                 context.restore();
+                // Get a map of step id to created resources
+                final Map<String, ResourceCreated> resourceMap = response.getWorkflowState()
+                    .resourcesCreated()
+                    .stream()
+                    .collect(Collectors.toMap(ResourceCreated::workflowStepId, Function.identity()));
```

Comment:
Consider appending step name to step ID with some delimiter, to make the pair unique?

Reply:
Answered above

Reply:
> Answered above

Yeah, I see that the map is OK, but looking at it all again, it seems to be overcomplicated.

We already have `response.getWorkflowState().resourcesCreated()` that we are using to create this map (it ends up as the `values()` after this change).

But later we are iterating the entry set just to extract that same iteration of `resourcesCreated`, which already has all the pieces we want (ID without having to modify it, step name/type, and resource id).

Reply:
I have removed the resource map from the deprovision workflow action and instead passed the list of resource created